doc/subsystems/networking/network-management-api.rst

```diff
@@ -35,26 +35,66 @@ To avoid extra cost, all :cpp:func:`net_mgmt()` calls are direct. Though this
 may change in a future release, it will not affect the users of this
 function.
 
-Listening to network event
-**************************
+Listening to network events
+***************************
 
 You can receive notifications on network events by registering a
-callback function and specifying an event mask used to match one or
-more events that are relevant.
+callback function and specifying a set of events used to filter when
+your callback is invoked.
 
 Two functions are available, :cpp:func:`net_mgmt_add_event_callback()` for
 registering the callback function, and
 :cpp:func:`net_mgmt_del_event_callback()`
-for unregistering. A helper function, :cpp:func:`net_mgmt_init_event_cb()`, can
+for unregistering a callback. A helper function,
+:cpp:func:`net_mgmt_init_event_callback()`, can
 be used to ease the initialization of the callback structure.
 
-When an event is raised that matches a registered event mask, the
+When an event occurs that matches a callback's event set, the
 associated callback function is invoked with the actual event
 code. This makes it possible for different events to be handled by the
 same callback function, if desired.
 
-See an example of registering callback functions and using the network
-management API in :file:`tests/net/mgmt/src/mgmt.c`.
+.. warning::
+
+   Event set filtering allows false positives. All callback handler
```

Comment:
I'd merge the sentence at the end here and suggest some clarity edits:

    Event set filtering allows false positives for events that have the same
    layer and layer code.  A callback handler function **must** check
    the event code (passed as an argument) against the specific network
    events it will handle, **regardless** of how many events were in the
    set passed to :cpp:func:`net_mgmt_init_event_callback()`.